### PR TITLE
rtl8721csm/bluetooth: fix ble assert issue

### PR DESCRIPTION
**Pull request by an4967**
- prevent mm_free and mq_timedsend in interrupt context

### CODE REVIEW

File: os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/board/common/src/hci_adapter.c [hunk 1]
```diff
@@ -36,6 +36,10 @@ bool hci_rtk_tx_cb(void)
 {
     if (p_hci_rtk->tx_buf != NULL)
     {
+        if (up_interruptcontext() == true) {
+            return 0;
+		}
```
Comment (sangam-swami) [inline os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/board/common/src/hci_adapter.c:41]: for proper indent, we can remove extra tab..

Comment (sangam-swami) [reply os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/board/common/src/hci_adapter.c:41]: we can use return false to maintain same format with others in this function

**Review by jeongchanKim — COMMENTED** [review]
Please check the indentation.

**Review by jeongchanKim — DISMISSED** [review]
LGTM

File: os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/board/common/src/hci_adapter.c [hunk 2]
```diff
@@ -34,7 +34,7 @@ extern bool hci_board_complete(void);
 /* ================================internal=========================== */
 bool hci_rtk_tx_cb(void)
 {
-    if (p_hci_rtk->tx_buf != NULL)
+    if (p_hci_rtk->tx_buf != NULL && up_interruptcontext() == false)
```
Comment (sunghan-chang) [inline os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/board/common/src/hci_adapter.c:37]: Is it ok not to free allocated memory? I think this makes memory leak.

Comment (sunghan-chang) [reply os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/board/common/src/hci_adapter.c:37]: I think you need to free it like `sched_free( )`.

Comment (sunghan-chang) [reply os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/board/common/src/hci_adapter.c:37]: It does not mean that you can call sched_free here. I mean you can refer the way.

Comment (sunghan-chang) [reply os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/board/common/src/hci_adapter.c:37]: Or Realtek should re-implement this logic not to call free in interrupt handler.

**Comment by an4967** [comment]
Realtek would fix this issue.

**Pull request closed without merging**